src/datatablerow/datatablerow.jl

```diff
@@ -37,31 +35,119 @@ Base.done(r::DataTableRow, s) = s > length(r)
 
 Base.convert(::Type{Array}, r::DataTableRow) = convert(Array, r.dt[r.row,:])
 
+Base.collect(r::DataTableRow) = Tuple{Symbol, Any}[x for x in r]
+
+# the equal elements of nullable and normal arrays would have the same hashes
+const NULL_MAGIC = 0xBADDEED # what to hash if the element is null
+
+# hash column element
+Base.@propagate_inbounds hash_colel(v::AbstractArray, i, h::UInt = zero(UInt)) = hash(v[i], h)
+Base.@propagate_inbounds hash_colel{T}(v::NullableArray{T}, i, h::UInt = zero(UInt)) =
+    isnull(v, i) ? hash(NULL_MAGIC, h) : hash(get(v[i]), h)
+Base.@propagate_inbounds hash_colel{T}(v::AbstractCategoricalArray{T}, i, h::UInt = zero(UInt)) =
+    hash(CategoricalArrays.index(v.pool)[v.refs[i]], h)
```

Comment:
It would be surprising that it would be significantly slower, since the code is very similar. Though since we need the special method for `NullableCategoricalArray` to avoid the cost of creating a `Nullable` just to unwrap it, I guess it doesn't matter to much what we do here.